lib/sidebar.rb

```diff
@@ -29,185 +29,193 @@ def pre_render(page)
+        rescue StandardError => e
+          Jekyll.logger.error("           Sidebar: Unable to render sidebar_markup for '#{filename}'.")
+          Jekyll.logger.debug("           Sidebar: #{e.message}. #{e.backtrace.inspect}")
+          next
+        end
+
+        sidebar_containers = doc.xpath('//*[@id="dx-sidebar-main-nav-ul"]')
+
+        unless sidebar_containers.any?
+          Jekyll.logger.debug("           Sidebar: No sidebar container found in #{filename}")
+          next
+        end
+
+        sidebar_containers.each do |sidebar_container|
+          sidebar_container.inner_html = sidebar_markup
+        end
 
-    def sanitize_filename(filename)
-      sanitized_filename = filename.match(/(?m)(?<=\b_site).*$/)[0]
-      sanitized_filename = sanitized_filename.gsub('index.html', '')
-      sanitized_filename.gsub('.html', '')
+        Jekyll.logger.debug("   Writing Sidebar: #{filename}")
+        File.open(filename, 'w') { |f| f.write(doc.to_html(encoding: 'UTF-8')) }
+      end
     end
 
-    def generateSubgroup(filename, key, value, all_subgroups, level)
-      title = value[:title].split('–').last
-      
-      subsubgroup_list = all_subgroups.select do |subsubgroup_key, _subsubgroup_value|
-        subsubgroup_key.include? key and subsubgroup_key != key and \
-          key.split('/').length > level
+    def generate_sub_group(filename, path, page, all_child_pages, level)
+      title = page[:title].split('–').last.strip
+
+      url = page[:url]
+      headers = page[:headers]
+
+      grandchildren = all_child_pages.select do |grandchild_path, _|
+        grandchild_path.include? path and grandchild_path != path and \
+          path.split('/').length > level
       end
 
-      subgroup = ''
-      has_subgroups = !all_subgroups.empty?
-      if value[:headers].any? || !subsubgroup_list.empty?
-        if has_subgroups
-          url = value[:url]
-          active = active?(filename, url, true)
-          # puts "#{url}, #{filename}, #{key}" if active
-          item_class = active || (url.split('/').length > level && filename.start_with?(url)) ? 'nav-subgroup active' : 'nav-subgroup'
-          subgroup << "<li class=\"#{item_class}\">"
-          subgroup << "<div class=\"nav-subgroup-heading\"><i class=\"material-icons\">arrow_right</i><a href=\"#{url}\">#{title}</a></div>"
-          subgroup << '<ul class="nav-ul">'
+      sub_group_markup = ''
+      has_child_pages = !all_child_pages.empty?
 
-          if subsubgroup_list.empty?
-            value[:headers].each do |header|
-              subgroup << "<li class=\"nav-leaf\"><a href=\"#{value[:url]}#{header[:hash]}\">#{header[:title]}</a></li>"
+      if (!headers.nil? && headers.any?) || !grandchildren.empty?
+        if has_child_pages
+          active = filename.active?(url, exact: true)
+          item_class = active || (url.split('/').length > level && filename.start_with?(url)) ? 'nav-subgroup active' : 'nav-subgroup'
```

Comment:
Codacy found an issue: [Line is too long. [134/80]](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-design-guide-jekyll-theme/pullRequest?prid=6198407)